### PR TITLE
CMR-3365: Fixed B and E parts of the ticket.

### PR DESCRIPTION
**Pull request by sxu123**
This PR includes fixes for part B, C, E, F, I. of CMR-3365.

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        depth-distance-units (get depth-distance-units-mapping
+                                  (when old-depth-distance-units
+                                    (string/upper-case old-depth-distance-units)))]
+      (-> c
+          (assoc-in [:SpatialInformation :VerticalCoordinateSystem :AltitudeSystemDefinition :DistanceUnits]  
+                    altitude-distance-units)
+          (assoc-in [:SpatialInformation :VerticalCoordinateSystem :DepthSystemDefinition :DistanceUnits]
+                    depth-distance-units)
+          ;; :SpatialInformation could become an empty map after setting DistanceUnits to nil
+          util/remove-empty-maps)))
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj]: I'd put this line and comment in the main migration function so we don't have to rely on the ordering of the migration functions.

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version_migration.clj [hunk 2]
```diff
@@ -1351,6 +1351,122 @@
                                                                           :Unit "dB"
                                                                           :Value "10"}]
                                                        :Technique "Drunken Fist"}]}]}]})))))
+  (testing "GeographicCoordinateUnits migration from version 1.9 to 1.10"
+  (is (= {:HorizontalCoordinateSystem
```
Comment (laurenfrederick) [inline umm-spec-lib/test/cmr/umm_spec/test/migration/version_migration.clj]: Looks like indentation is off here. The `(is ...)` blocks should be indented under the `(testing )` block

File: umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  {"FATHOMS" "Fathoms"
+   "FEET" "Feet"
+   "HECTOPASCALS" "HectoPascals"
+   "METERS" "Meters"
+   "MILLIBARS" "Millibars"})
+
+(defn migrate-distance-units-to-enum
+  "Migrate distance units from string to enum."
+  [c]
+  (let [old-altitude-distance-units (-> c
```
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj]: Is there any reason these threads couldn't be a `get-in`? I noticed this in a few places

File: umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                                        :SpatialInformation
+                                        :VerticalCoordinateSystem
+                                        :AltitudeSystemDefinition
+                                        :DistanceUnits)
+        altitude-distance-units (get altitude-distance-units-mapping
+                                     (when old-altitude-distance-units
```
[the remark this answers is not in the document]
Comment (sxu123) [reply umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj]: Good idea. Done.

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version_migration.clj (hunk 2 above)
Comment (sxu123) [reply umm-spec-lib/test/cmr/umm_spec/test/migration/version_migration.clj]: Done.

File: umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj (hunk 3 above)
Comment (sxu123) [reply umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj]: No reason. just a habit. Will change to get-in. Done.

File: umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj (hunk 1 above)
Comment (sxu123) [reply umm-spec-lib/src/cmr/umm_spec/migration/distance_units_migration.clj]: Done.

**Pull request merged**